cogs/chal/chal.py

```diff
@@ -1,167 +1,66 @@
-    @subcommand_decorator(name={'description': "The name of the challenge to create"})
-    @run_from_ctf
-    async def create(self, ctx: SlashContext, name: str) -> None:
-        """
-            Creates a new challenge inside a CTF
-        """
+from lib.util import subcommand, get_ctf_forum
+from lib.config import FORUM_GENERAL_CHANNEL
 
-        await ctx.defer()
-        # Get CTF category called in
-        category_folder = ctx.channel.category
 
-        channel_name = sanitize_channel_name(name)
+class Chal(Extension):
+    '''Commands for managing challenges in a CTF forum'''
 
-        if channel_name in AUTOGENERATED_CHANNELS.values():
-            await ctx.send('Cannot create a challenge with the same name as an autogenerated channel.')
+    @subcommand(name={"description": "The name of the challenge"}, category={"description": "the category of the challenge", "autocomplete": True})
+    async def create(self, ctx: SlashContext, name: str, category: str) -> None:
+        '''Creates a channel for the challenge'''
+        forum = await get_ctf_forum(ctx)
+        if (forum == None or ctx.channel.name != FORUM_GENERAL_CHANNEL):
+            await ctx.send(":x: Must be used inside a CTF forum's general channel.")
             return
-        for ctf_channel in category_folder.text_channels:
-            if(ctf_channel.name == channel_name):
-                await ctx.send('Cannot create a challenge with the same name as an existing channel.')
+        if (category == "unsolved"):
+            await ctx.send(":x: Unsolved cannot be a category.")
+            return
+        for post in forum.get_posts(exclude_archived=False):
+            if ((post.name or "") == name):
+                await ctx.send(f":x: Challenge {name} already exists.")
                 return
-        await ctx.guild.create_text_channel(
-            name=channel_name,
-            category=category_folder,
-        )
-
-        # Get bot channel and send embed
-
-        embed = discord.Embed(
-            title="🔔 New challenge created!",
-            description=(
-                f"**Challenge name:** {name}\n"
-            ),
-            colour=discord.Colour.dark_gold(),
-        ).set_footer(text=datetime.strftime(datetime.now(), "%a, %d %B %Y, %H:%M UTC"))
-        await ctx.send(embed=embed)
-
-    @subcommand_decorator(flag={'description': "The flag for this challenge."})
-    @run_from_ctf
-    async def solve(self, ctx: SlashContext, flag: str) -> None:
-        """Solves a CTF challenges
-
-        """
-        await ctx.defer()
-        challenge = ctx.channel
 
-        if challenge.name in AUTOGENERATED_CHANNELS.values():
-            await challenge.send(f'You cannot solve the channel {challenge.name}!')
+        # check valid category
+        for tag in forum.available_tags:
+            if (tag.name == category):
+                break
+        else:  # this indentation is intentional
```

Comment:
I don't like this coding style, can you just do

`any(x.name == category for x in forum.available_tags)`